sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py

```diff
@@ -62,8 +73,6 @@ def __init__(self, dataflow_client=None, job_result=None, job_graph=None):
         dataflow service.
       job_result: DataflowPipelineResult with the state and id information of
         the job.
-      job_graph: apiclient.Job instance to be able to translate between internal
-        step names (e.g. "s2"), and user step names (e.g. "split").
```

Comment:
I see the Arg description was removed, but the argument itself wasnt removed. Maybe leave the description, and clarify that it's optional?

What do you think about simply removing it completely from the class? (instead receiving the job id only) The difference would be that we lose the ability to cache metrics... but perhaps that's acceptable. Thoughts?